### PR TITLE
Added possibility to toggle search bar input

### PR DESCRIPTION
**Pull request by JNussens**
#### What problem is this solving?
Users cant collapse the search bar input

Users should be able to choose if they want the search bar to be collapsible or not
#### How should this be manually tested?
Just click on the search-bar search icon and see if it toogles the input when using toggleableInput prop.
https://dev--miriadeit.myvtex.com/

#### Checklist/Reminders

- [x] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements 

#### Notes

For project Miriade (Daniel Vianna).

### CODE REVIEW

**Comment by klzns** [comment]
Thanks for the pull request. Can you provide a workspace with this feature linked?

**Comment by JNussens** [comment]
Yes, of course: [https://dev--miriadeit.myvtex.com/](url) Please test on mobile. 

Thanks @klzns !

**Comment by klzns** [comment]
I think it didn't work:

Jan-07-2020 15-57-15

**Comment by JNussens** [comment]
@klzns Oh, I just fixed this, sorry, I was developing other features while linked. Please check now

File: react/components/SearchBar/components/AutocompleteInput.js [hunk 1]
```diff
@@ -64,13 +66,20 @@ const AutocompleteInput = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
 
+  const toggleInput = () => {
```
Comment (klzns) [inline react/components/SearchBar/components/AutocompleteInput.js]:
```suggestion
  const handleClick = () => {
```

File: react/components/SearchBar/components/AutocompleteInput.js [hunk 2]
```diff
@@ -64,13 +66,20 @@ const AutocompleteInput = ({
...
+  }
+
   const suffix = (
     <button
       className={`${iconClasses || ''} ${
         handles.searchBarIcon
       } flex items-center pointer bn bg-transparent outline-0`}
       onClick={
-        submitOnIconClick ? onGoToSearchPage : () => value && onClearInput()
+        toggleInput
```
Comment (klzns) [inline react/components/SearchBar/components/AutocompleteInput.js]:
```suggestion
        handleClick
```

File: docs/SearchBar.md [hunk 3]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (victorhmp) [inline docs/SearchBar.md]:
```suggestion
```

File: react/components/SearchBar/index.js [hunk 4]
```diff
@@ -56,6 +56,7 @@ class SearchBarContainer extends Component {
       iconClasses,
       autoFocus,
       maxWidth,
+      toggleableInput,
```
Comment (victorhmp) [inline react/components/SearchBar/index.js]:
```suggestion
      toggleableInput = false,
```

Comment (victorhmp) [reply react/components/SearchBar/index.js]: This is just to make it clearer

**Comment by JNussens** [comment]
@victorhmp I've fixed it, it's less weird now. I think it's suitable to merge

**Comment by victorhmp** [comment]
> @victorhmp I've fixed it, it's less weird now. I think it's suitable to merge

It is much better! But just one more thing, regarding this fade animation. Wouldn't it be better if there were a prop to toggle it?

**Comment by JNussens** [comment]
> > @victorhmp I've fixed it, it's less weird now. I think it's suitable to merge
> 
> It is much better! But just one more thing, regarding this fade animation. Wouldn't it be better if there were a prop to toggle it?

Nice! In my opinion, I think this feature should bundle with the animation, since without it, the feature presents it-self very clunky, so it should be a standard having the feature animated. What do you think?

**Comment by klzns** [comment]
I created a workspace with this feature linked:
https://brenosearch--storecomponents.myvtex.com/

File: react/components/SearchBar/styles.css [hunk 5]
```diff
@@ -34,3 +34,12 @@
     box-shadow: none;
   }
 }
+
+.fadeIn {
+  animation: fadeIn .3s ease-in-out;
```
Comment (lbebber) [inline react/components/SearchBar/styles.css]: i think this transition is a bit too slow. 150ms would be better I think

File: react/components/SearchBar/components/AutocompleteInput.js [hunk 6]
```diff
@@ -94,16 +104,33 @@ const AutocompleteInput = ({
...
+                  ref={inputRef}
+                  size="large"
+                  value={value}
+                  prefix={hasIconLeft && prefix}
+                  suffix={suffix}
+                  {...restProps}
+                  error={Boolean(inputErrorMessage)}
+                  errorMessage={inputErrorMessage}
+                />
+              </div> : suffix
```
Comment (hiagolcm) [inline react/components/SearchBar/components/AutocompleteInput.js]: When the `toggleInput`changes from `true` to `false`, the Input is not rendered anymore.  This way, there is no animation when the user closes the input. 

What do you think about to add an animation in both situations: opening and closing?

**Comment by victorhmp** [comment]
Closing this since it became stale.
@JNussens feel free to reopen this if you still plan on working in this feature.

**Pull request closed without merging**